### PR TITLE
fixing docstrings and load_simulation

### PR DESCRIPTION
**Pull request by Lilyhanig**
- fixing docstrings
- fixing issues with load_simulation_results
- changing a few function names
- adding the ability to write to a CSV and to have the time_stamp in the variable files
- allowing the ability to plot AggregatedResults struct

### CODE REVIEW

File: src/core/core_structs/operations_problem.jl [hunk 1]
```diff
@@ -68,15 +68,15 @@ OpModel = OperationsProblem(TestOpProblem, template, c_sys5_re; PTDF = PTDF5, op
 ```
 
 # Accepted Key Words
--`verbose::Bool = true`: verbose default is true
--`PTDF::PTDF = PTDF`: Passes the PTDF matrix into the optimization model
--`optimizer::union{Nothing,JuMP.OptimizerFactory} = GLPK_optimizer`: The optimizer gets passed
+- `verbose::Bool = true`: verbose default is true
+- `PTDF::PTDF = PTDF`: Passes the PTDF matrix into the optimization model
+- `optimizer::union{Nothing,JuMP.OptimizerFactory} = GLPK_optimizer`: The optimizer gets passed
```
Comment (jd-lara) [inline src/core/core_structs/operations_problem.jl]: Don't include there GLPK_optimizer, that is specific to a use case.

File: src/core/core_structs/operations_problem.jl [hunk 2]
```diff
@@ -104,12 +104,12 @@ This uses the Abstract Power Formulation to build the model reference and
 the optimization model and populates the operation model struct.
 
 # Arguments
--`op_problem::Type{M} = where {M<:AbstractOperationsProblem`: Defines the type of the operation model
--`::Type{T} where T<:PM.AbstractPowerFormulation`: The power formulation used for model ref & optimization model
--`sys::PSY.System = c_sys5`: the system created in Power Systems
+- `op_problem::Type{M} = where {M<:AbstractOperationsProblem`: Defines the type of the operation model
+- `::Type{T} where T<:PM.AbstractPowerFormulation`: The power formulation used for model ref & optimization model
+- `sys::PSY.System = c_sys5`: the system created in Power Systems
```
Comment (jd-lara) [inline src/core/core_structs/operations_problem.jl]: same here don't include c_sys5 that's specific to an example

File: src/core/core_structs/operations_problem.jl [hunk 3]
```diff
@@ -120,15 +120,15 @@ OpModel = OperationsProblem(TestOpProblem, template, c_sys5_re; PTDF = PTDF5, op
...
+- `verbose::Bool = true`: verbose default is true
+- `PTDF::PTDF = PTDF`: Passes the PTDF matrix into the optimization model
+- `optimizer::union{Nothing,JuMP.OptimizerFactory}`: The optimizer gets passed
 into the optimization model the default is nothing.
--`initial_conditions::DICKDA = DICKDA()`: default of Dict{ICKey, Array{InitialCondition}}
--`parameters::Bool = false`: enable JuMP parameters
--`use_forecast_data::Bool = true`: if true, forecast collects the time steps in Power Systems,
+- `initial_conditions::DICKDA = DICKDA()`: default of Dict{ICKey, Array{InitialCondition}}
+- `parameters::Bool = false`: enable JuMP parameters
+- `use_forecast_data::Bool = true`: if true, forecast collects the time steps in Power Systems,
```
Comment (jd-lara) [inline src/core/core_structs/operations_problem.jl]: same here, don't include = true or = false

File: src/core/core_structs/operations_problem.jl [hunk 4]
```diff
@@ -170,15 +170,15 @@ OpModel = OperationsProblem(TestOpProblem, template, c_sys5_re; PTDF = PTDF5, op
 ```
 
 # Accepted Key Words
--`verbose::Bool = true`: verbose default is true
--`PTDF::PTDF = PTDF`: Passes the PTDF matrix into the optimization model
--`optimizer::union{Nothing,JuMP.OptimizerFactory}`: The optimizer gets passed
+- `verbose::Bool = true`: verbose default is true
+- `PTDF::PTDF = PTDF`: Passes the PTDF matrix into the optimization model
```
Comment (jd-lara) [inline src/core/core_structs/operations_problem.jl]: same here

File: src/core/core_structs/results.jl [hunk 5]
```diff
@@ -84,3 +103,33 @@ function shorten_time_stamp(time::DataFrames.DataFrame)
     time = time[1:(size(time,1)-1),:]
     return time
 end
+
+function load_operation_results(folder_path::AbstractString, file_type)
+
+    if isfile(folder_path)
+        @error("not a folder path")
```
Comment (jd-lara) [inline src/core/core_structs/results.jl]: This doesn't seem to be the correct type of error. cc. @daniel-thom

File: src/routines/write_results.jl [hunk 6]
```diff
@@ -162,25 +144,48 @@ function _export_optimizer_log(optimizer_log::Dict{Symbol, Any},
...
+    write_model_results(results::Results, save_path::String)
+
+Exports Operational Problem Results to a path
+
+# Arguments
+- `results::Results`: results from the simulation
+- `save_path::String`: folder path where the files will be written
+
+# Accepted Key Words
+- `file_type = CSV`: only CSV and featherfile are accepted
```
Comment (jd-lara) [inline src/routines/write_results.jl]: same here

File: src/routines/write_results.jl [hunk 7]
```diff
@@ -162,25 +144,48 @@ function _export_optimizer_log(optimizer_log::Dict{Symbol, Any},
...
+"""
+function write_results(res::AggregatedResults, path::String, results_folder::String; kwargs...)
     if !isdir(path)
         error("Specified path is not valid. Run write_results to save results.")
     end
-    folder_path = joinpath(path,results)
+    folder_path = joinpath(path,results_folder)
```
Comment (jd-lara) [inline src/routines/write_results.jl]: add space after the comma

File: src/utils/aggregation.jl [hunk 8]
```diff
@@ -155,17 +155,19 @@ variable = [:P_ThermalStandard, :P_RenewableDispatch]
...
 ```
 # Accepted Key Words
--`write::Bool`: if true, the aggregated results get written back to the results
+- `write::Bool`: if true, the aggregated results get written back to the results
 file in the folder structure
+- `file_type::String = CSV`: default filetype is Feather, but this key word can be used to make it CSV.
+if a different file type is desired the code will have to be changed to accept it.
 """
 function load_simulation_results(stage::String,
          step::Array,
```
Comment (jd-lara) [inline src/utils/aggregation.jl]: alignment is off.

File: src/utils/aggregation.jl [hunk 9]
```diff
@@ -155,17 +155,19 @@ variable = [:P_ThermalStandard, :P_RenewableDispatch]
 results = load_simulation_results(stage,step, variable, references)
 ```
 # Accepted Key Words
--`write::Bool`: if true, the aggregated results get written back to the results
+- `write::Bool`: if true, the aggregated results get written back to the results
 file in the folder structure
+- `file_type::String = CSV`: default filetype is Feather, but this key word can be used to make it CSV.
+if a different file type is desired the code will have to be changed to accept it.
 """
 function load_simulation_results(stage::String,
```
Comment (jd-lara) [inline src/utils/aggregation.jl]: Have added tests for this function? Looks a bit fragile

File: src/core/core_structs/results.jl (hunk 5 above)
Comment (daniel-thom) [reply src/core/core_structs/results.jl]: Correct, that is an error log print only.  If you want to raise an exception then call `error()`

**Pull request closed without merging**